server/src/assets/game/tableGenerator/converter.py

```diff
@@ -1,127 +1,184 @@
+        if configuration in flagMappingFileNames:
+            configNamesWithCSVList.append(configuration)
+    return configNamesWithCSVList
+
+
+def getSignalIDStrsWithDefinedMapping(signalFlagCSVFilepath: str, extrasConfig) -> list[str]:
+    """
+    Reads the signal-to-symbol/flag CSV file and extracts all signal IDs (represented as strings,
+    in the long format, e.g., `signal8`). If the signal is actually a composite signal, then its
+    ID is appendend an "a" as all composite signals AT THE MOMENT have the "proper" (non-distant)
+    signal ID with an "a". (e.g., signal4 will be signal4a in the returned list)
+    """
+    signalIDList = []
+    with open(signalFlagCSVFilepath, "r") as csvFile:
         csv_reader = reader(csvFile)
         for row in csv_reader:
             signal = "signal" + row[0]
-            if signal in destinations:
-                destinationsSorted.append(signal)
-            signal = signal + "a"
-            if signal in destinations:
-                destinationsSorted.append(signal)
-
-    resultData[block] = {}
-    for destination in destinationsSorted:
-        resultData[block][destination] = {}
-        resultData[block][destination]["route-id"] = originalResultData[block][destination]["route-id"]
-        resultData[block][destination]["orientation"] = originalResultData[block][destination]["orientation"]
-        resultData[block][destination]["block"] = originalResultData[block][destination]["block"]
-        resultData[block][destination]["segment"] = originalResultData[block][destination]["segment"]
-
-with open("../destinations-swtbahn-full.json", "w") as file:
-    file.write("const allPossibleDestinationsSwtbahnFull = ")
-    file.write(json.dumps(resultData, indent=2))
-    file.write(";")
+            if extrasConfig["compositions"] != None:
+                for comp in extrasConfig["compositions"]:
+                    if signal == comp["id"]:
+                        signal += "a"
+            signalIDList.append(signal)
+    return signalIDList
+
+
+def getBlacklistedRouteIDsForConfig(config: str) -> list[int]:
+    """
+    Get a list of all route IDs blacklisted for this configuration/model railway platform.
+    If no blacklist file for the config is found, an empty list is returned.
+    """
+    global blacklistFileDirectory
+    blacklistedRouteIDs = []
+    if os.path.exists("{}/{}.txt".format(blacklistFileDirectory, config)):
+        for line in open("{}/{}.txt".format(blacklistFileDirectory, config)):
+            blacklistedRouteIDs.append(int(line.strip()))
+    return blacklistedRouteIDs
+
+
+def getInterlockingAndExtrasConfigFileContent(config: str) -> any:
+    """
+    Attempts to load the interlocking table and extras-config configuration files into dicts.
+    Throws an exception if the config files are not found.
+    """
+    global pathToConfig, interlockingTableFileName, extrasConfigFileName
+    # Det. filepaths for config files and load them
```

Comment:
What is Det. short for?

Reply:
Adjusted the comment to `Determine filepaths for config files, then load the config file content`